Client/mods/deathmatch/logic/luadefs/CLuaEffectDefs.cpp

```diff
@@ -498,6 +499,58 @@ int CLuaEffectDefs::fxAddFootSplash(lua_State* luaVM)
     return 1;
 }
 
+int CLuaEffectDefs::FxAddShadow(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    eShadowType      shadowType = SHADOW_NONE;
+    CVector          vecPosition;
+    CVector2D        vecOffset1;
+    CVector2D        vecOffset2;
+
+    SColor color;
+    float  zDistance;
+    bool   bDrawOnWater;
+    bool   bDrawOnBuildings;
+
+    if (argStream.NextIsString())
+        argStream.ReadEnumString(shadowType);
+    argStream.ReadVector3D(vecPosition);
+    argStream.ReadVector2D(vecOffset1);
+    argStream.ReadVector2D(vecOffset2);
+    argStream.ReadColor(color, SColor(0x80FFFFFF));
+    argStream.ReadNumber(zDistance, 4);
+    argStream.ReadBool(bDrawOnWater, false);
+    argStream.ReadBool(bDrawOnBuildings, false);
+    if (!argStream.HasErrors())
+    {
+        if (vecOffset1.Length() > 32)
+        {
+            argStream.SetCustomError("First offset can not be longer than 32 units");
+        }
+        else if (vecOffset2.Length() > 32)            // bigger and close to limit shadows size can be partially invisible
+        {
+            argStream.SetCustomError("Second offset can not be longer than 32 units");
+        }
+        else if (zDistance < 0 || zDistance > 3000)            // negative distance not working
```

Comment:
no changes needed from you here @CrosRoad95, just dropping a note for @sbx320 - can we have a ranged number type in #973 that automates this check?